packages/db/src/provider.ts

```diff
@@ -0,0 +1,59 @@
+import omit from 'lodash/omit';
+import isNil from 'lodash/isNil';
+import isValidHostname from 'is-valid-hostname';
+
+import { ProviderRecord, PostgresProviderRecord } from '@cumulus/types';
+import { PostgresValidationError } from '@cumulus/errors';
+
+import KMS from '@cumulus/aws-client/KMS';
+
+export const encryptValueWithKMS = (
+  value: string
+): Promise<string> => {
+  if (process.env?.provider_kms_key_id === undefined) {
+    throw new Error('env variable provider_kms_key_id must be set');
+  }
+  return KMS.encrypt(process.env?.provider_kms_key_id, value);
+};
+
+export const postgresProviderFromCumulusProvider = async (
+  data: ProviderRecord,
+  encryptMethod: Function = encryptValueWithKMS
+) => {
+  let username: string | undefined;
+  let password: string | undefined;
+  if (data.username) {
```

Comment:
This is similar to the logic in the providers migration:

https://github.com/nasa/cumulus/blob/feature-rds/lambdas/data-migration1/src/providers.ts#L76

The difference is that the providers Dynamo -> RDS migration has to account for values from Dynamo being encrypted with the S3 Keypair provider, whereas your case is just taking in raw unencrypted values when creating records. So it's probably best to keep the logic separate

Reply:
Agreed.